src/main/java/org/spongepowered/api/network/packet/PacketChannel.java

```diff
@@ -0,0 +1,193 @@
+
+    /**
+     * Gets the {@link PacketBinding} for the given packet class, if the packet
+     * type is registered to this channel.
+     *
+     * @param packetClass The packet class
+     * @param <M> The type of the packet
+     * @return The packet binding, if found
+     */
+    <M extends Packet> Optional<PacketBinding<M>> getBinding(Class<M> packetClass);
+
+    /**
+     * Register a request/response packet class pair to this channel. A receiving handler
+     * isn't required when sending the request using {@link #sendTo(ClientConnection, RequestPacket)}.
+     *
+     * <p>The packet opcode is used to identify this packet class as it is
+     * sent and received across the network, it is encoded as a varint.</p>
+     *
+     * @param packetOpcode A unique opcode for this request/response packet pair
+     * @param requestPacketClass The class of the request packet being registered
+     * @param responsePacketClass The class of the response packet being registered
+     * @return The created transactional packet binding
+     */
+    <M extends RequestPacket<R>, R extends ResponsePacket> TransactionalPacketBinding<M, R> register(
+            Class<M> requestPacketClass, Class<R> responsePacketClass, int packetOpcode);
+
+    /**
+     * Gets the {@link TransactionalPacketBinding} for the given {@link RequestPacket} class, if
+     * the type is registered to this channel as a transactional binding.
+     *
+     * @param requestPacket The request packet class
+     * @param <M> The type of the request packet
+     * @param <R> The type of the response packet
+     * @return The transactional packet binding, if found
+     */
+    <M extends RequestPacket<R>, R extends ResponsePacket> Optional<TransactionalPacketBinding<M, R>> getTransactionalBinding(
+            Class<M> requestPacket);
+
+    /**
+     * Sends the packet to the player using this channel. The packet may
+     * not be sent if the player doesn't have a registered handler.
+     *
+     * @param player The player to send the packet to
+     * @param packet The packet to send
+     */
+    default void sendTo(Player player, Packet packet) {
+        sendTo(player.getConnection(), packet);
+    }
+
+    /**
+     * Sends the {@link Packet} to the client using this channel.
+     *
+     * <p>A exception will be thrown if the specified packet type
+     * isn't registered in this {@link PacketChannel}.</p>
+     *
+     * @param connection The player connection to send the packet to
+     * @param packet The packet to send
+     */
+    void sendTo(ClientConnection connection, Packet packet);
+
+    /**
+     * Sends the {@link Packet} to the player using this channel. This method returns a
+     * {@link CompletableFuture} that will be accepted once the response is received.
+     *
+     * <p>The {@link CompletableFuture} may fail exceptionally by a
+     * {@link NoResponseException} if there wasn't a valid response
+     * received for the given request.</p>
```

Comment:
How is this determined? Is there some kind of timeout?

Reply:
Timeout or a "no response" packet.